src/qtils/error.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <any>
+#include <cstdint>
+#include <deque>
+#include <system_error>
+
+#include <fmt/format.h>
+
+#include <qtils/optref.hpp>
+
+#define Q_ERROR_LOCATION \
+  (::qtils::ErrorLocation{__PRETTY_FUNCTION__, __FILE__, __LINE__})
+
+#define Q_ERROR(error_) (::qtils::Error{Q_ERROR_LOCATION, error_})
```

Comment:
```suggestion
#define Q_ERROR(error_) (::qtils::Error{Q_ERROR_LOCATION, (error_)})
```

Reply:
Not an error here (`ADD(a,b)a+b; ADD(2*3,4*5); 2*3+4*5`)